File: examples/ruby/user_service.rb
# Example Ruby file with logging patterns
require 'logger'

class UserService
  def initialize
    @logger = Logger.new(STDOUT)
  end

  def get_user(user_id)
    @logger.info "Fetching user with ID: #{user_id}"

    if user_id < 0
      @logger.error "Invalid user ID: #{user_id}"
      return nil
    end

    if user_id == 404
      @logger.warn "User not found in database"
      return nil
    end

    @logger.debug "Successfully retrieved user #{user_id}"
    { id: user_id, name: 'John Doe' }
  end

  def create_user(username, email)
    @logger.info "Creating new user: #{username}"

    if email.nil? || email.empty?
      @logger.error "Email is required"
      raise ArgumentError, "Email is required"
    end

    user = { username: username, email: email }
    @logger.debug "User created successfully: #{user}"

    user
  end
end

# Rails-style logging
Rails.logger.info "Application starting" if defined?(Rails)

File: examples/javascript/user-service.js
// Example JavaScript file with logging patterns
const winston = require('winston');

const logger = winston.createLogger({
  level: 'info',
  format: winston.format.json(),
  transports: [new winston.transports.Console()],
});

class UserService {
  constructor() {
    this.logger = logger;
  }

  getUser(userId) {
    this.logger.info(`Fetching user with ID: ${userId}`);

    if (userId < 0) {
      this.logger.error(`Invalid user ID: ${userId}`);
      return null;
    }

    if (userId === 404) {
      this.logger.warn('User not found in database');
      return null;
    }

    this.logger.debug(`Successfully retrieved user ${userId}`);
    return { id: userId, name: 'John Doe' };
  }

  createUser(username, email) {
    this.logger.info(`Creating new user: ${username}`);

    if (!email) {
      this.logger.error('Email is required');
      throw new Error('Email is required');
    }

    const user = { username, email };
    this.logger.debug(`User created successfully: ${JSON.stringify(user)}`);

    return user;
  }
}

// Console logging
console.log('Application starting');
console.error('This is an error message');
console.warn('This is a warning message');

module.exports = UserService;
